lib/reporters/default.js

```diff
@@ -0,0 +1,1125 @@
+		}
+
+		this._writeWithSpinner(last.chunk.toString('utf8'));
+		callback();
+	}
+
+	_writeWithSpinner(string) {
+		if (!this.spinner) {
+			return;
+		}
+
+		if (!this.spinner.id) {
+			this.dest.write(string);
+			return;
+		}
+
+		this.lastSpinnerText = string;
+		// Ignore whitespace at the end of the chunk. We're continiously rewriting
+		// the last line through the spinner. Also be careful to remove the indent
+		// as the spinner adds its own.
+		this.spinner.text = string.trimEnd().slice(2);
+		this.spinner.render();
+	}
+
+	writeLine(string) {
+		if (string) {
+			this.write(indentString(string, 2) + os.EOL);
+			this.lastLineIsEmpty = false;
+		} else {
+			this.write(os.EOL);
+			this.lastLineIsEmpty = true;
+		}
+	}
+
+	ensureEmptyLine() {
+		if (!this.lastLineIsEmpty) {
+			this.writeLine();
+		}
+	}
+}
+
+class Reporter {
+	constructor({
+		isVerbose,
+		reportStream,
+		stdStream,
+		projectDir,
+		watching,
+		spinner,
+		durationThreshold
+	}) {
+		this.isVerbose = isVerbose;
+		this.reportStream = reportStream;
+		this.stdStream = stdStream;
+		this.watching = watching;
+		this.relativeFile = file => path.relative(projectDir, file);
+		this.consumeStateChange = whileCorked(
+			this.reportStream,
+			this.consumeStateChange
+		);
+
+		if (this.isVerbose) {
+			this.durationThreshold = durationThreshold || 100;
+			this.endRun = whileCorked(this.reportStream, this.endRun);
+			this.lineWriter = new LineWriter(this.reportStream);
+		} else {
+			this.spinner = ora({
+				isEnabled: true,
+				color: spinner ? spinner.color : 'gray',
+				discardStdin: !watching,
+				hideCursor: false,
+				spinner: spinner || (process.platform === 'win32' ? 'line' : 'dots'),
+				stream: reportStream
+			});
+			this.lineWriter = new LineWriter(this.reportStream, this.spinner);
+			this.endRun = whileCorked(
+				this.reportStream,
+				whileCorked(this.lineWriter, this.endRun)
+			);
+		}
+
+		this.reset();
+	}
+
+	reset() {
+		if (this.removePreviousListener) {
+			this.removePreviousListener();
+		}
+
+		this.prefixTitle = (testFile, title) => title;
+
+		this.runningTestFiles = new Map();
+		this.filesWithMissingAvaImports = new Set();
+		this.filesWithoutDeclaredTests = new Set();
+		this.filesWithoutMatchedLineNumbers = new Set();
+
+		this.failures = [];
+		this.internalErrors = [];
+		this.knownFailures = [];
+		this.lineNumberErrors = [];
+		this.uncaughtExceptions = [];
+		this.unhandledRejections = [];
+
+		this.previousFailures = 0;
+
+		this.failFastEnabled = false;
+		this.lastLineIsEmpty = false;
+		this.matching = false;
+
+		this.removePreviousListener = null;
+		this.stats = null;
+	}
+
+	startRun(plan) {
+		if (plan.bailWithoutReporting) {
+			return;
+		}
+
+		this.reset();
+
+		this.failFastEnabled = plan.failFastEnabled;
+		this.matching = plan.matching;
+		this.previousFailures = plan.previousFailures;
+		this.emptyParallelRun = this.isVerbose ?
+			plan.status.emptyParallelRun :
+			this.emptyParallelRun;
```

Comment:
This coupling against `verbose` is surprising. Yes, parallel runs only run in CI, and that's when the reporter is in `verbose` mode, but we should either always use `plan.status.emptyParallelRun` or have an option for this.